src/org/opendatakit/briefcase/ui/automation/AutomationPanel.java

```diff
@@ -0,0 +1,97 @@
+package org.opendatakit.briefcase.ui.automation;
+
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+import static org.opendatakit.briefcase.util.FindDirectoryStructure.isWindows;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.swing.JPanel;
+import org.bushe.swing.event.annotation.AnnotationProcessor;
+import org.opendatakit.briefcase.automation.AutomationConfiguration;
+import org.opendatakit.briefcase.export.ExportConfiguration;
+import org.opendatakit.briefcase.model.BriefcasePreferences;
+import org.opendatakit.briefcase.reused.BriefcaseException;
+import org.opendatakit.briefcase.ui.export.ExportPanel;
+import org.opendatakit.briefcase.ui.export.components.ConfigurationDialog;
+import org.opendatakit.briefcase.util.FormCache;
+
+public class AutomationPanel {
+  public static final String TAB_NAME = "Automation";
+
+  private final AutomationPanelForm view;
+  private final FormCache formCache;
+  private final BriefcasePreferences appPreferences;
+
+  public AutomationPanel(AutomationPanelForm view, BriefcasePreferences appPreferences, FormCache formCache) {
+    AnnotationProcessor.process(this);
+    this.view = view;
+    this.formCache = formCache;
+    this.appPreferences = appPreferences;
+
+    view.onGenerate(config -> {
+      generateScript(isWindows() ? "automation.bat" : "automation.sh", config);
+    });
+
+    view.onSetExportConfiguration(() -> {
```

Comment:
We shouldn't add this here, since the only callback we want to use from the view is `onGenerate()`

Reply:
That means we don't even need a call back for this right? Just a method in the view that sets the export configuration?

Reply:
One question. This is supposed to update the AutomationConfiguration.exportConfiguration field but AutomationConfiguration is not field of this class. It's always created inside the onGenerate callback. So what do I do with ExportConfiguration returned by the dialog?

Reply:
That’s correct: no callback needed

Reply:
You can store the conf object in a member of the AutomationPanelForm, inside an Optional object